src/wms/src/main/java/org/geoserver/wms/WMSServiceException.java

```diff
@@ -0,0 +1,133 @@
+/* (c) 2015 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.wms;
+
+import org.geoserver.platform.ServiceException;
+
+/**
+ * Extension of ServiceException for returning a WebMap with the exception.
+ * Used with WMS option EXCEPTIONS=PARTIALMAP to return a partial image if there is an 
+ * exception (such as a timeout) thrown when rendering a WMS request.
+ */
+public class WMSServiceException extends ServiceException {
```

Comment:
When Gabriel made the giant WMS refactor (2010?) he removed this class, which was present, under the notion that a single ServiceException was enough for everything. I did not agree, but at least the module is uniform now, if you want to re-introduce this class, please let's discuss it on the ml, and let's keep the module and its dependencies (kml) uniform (wmsexception everywhere or serviceexception everywhere)